src/app/@ansyn/plugins/openlayers/open-layers-north-calculations/plugin/north-calculations-plugin.ts

```diff
@@ -71,32 +84,92 @@ export class NorthCalculationsPlugin implements IMapPlugin {
 		});
 	}
 
-// override this method
-	projectPoints(iMap: IMap<OLMap>, coordinates: Coordinate[]): Observable<Point[]> {
+	// override this method
+	projectPoints(coordinates: Coordinate[]): Observable<Point[]> {
 		const observables = [];
 
 		coordinates.forEach(coordinate => {
 			const point = <GeoJSON.Point> turf.geometry('Point', coordinate);
-			observables.push(iMap.projectionService.projectAccurately(point, iMap));
+			observables.push(this.ActiveMap.projectionService.projectAccurately(point, this.ActiveMap));
 		});
 
 		return Observable.forkJoin(observables);
 	}
 
-	setCorrectedNorth(iMap, retryNumber = 0): Promise<number> {
+	setCorrectedNorth(retryNumber = 0): Promise<number> {
```

Comment:
Never really liked this code. Let's return an observable and use the `retry` operator. It will fit better since you already consume it via effects.